### ISSUE TITLE
PDF representation of the DGC

### ISSUE THREAD

**Issue by PeterEbraert** [ehn-digital-green-development/ehn-dgc-schema#40]
Hello all,

We are going to generate also a PDF representation of the DGC. We have many non-digital citizens, which we hope to reach with such representations. Anybody has experiences to share with respect to the look and feel of paper DGCs? 

Are there any European specifications or directives concerning the look and feel of the digital green card, for the PDF and paper version? Will there be specifics concerning the format, pixels, design,... Or does every issuer just have to make sure that the required fields are present and is the layout the responsibility of the issuer?

Thanks in advance,

Peter

**Comment by jschlyter** [comment]
Although this is an important question, I believe it is out of scope for the schema work. Perhaps @gabywh can give some advise what table to bring this issue to?

**Comment by gabywh** [comment]
Page 3 in https://ec.europa.eu/health/sites/health/files/ehealth/docs/trust-framework_interoperability_certificates_en.pdf suggests a format. Helpful, or were you looking for something different?

**Comment by gabywh** [comment]
> Or does every issuer just have to make sure that the required fields are present

That's the legal basis, for sure. As long as this is met, it will be fine.

I am not aware of any UX/UI specific layout criteria at EU level (e.g. so many pixels wide, so much % screen etc) but I'll ask around.

**Comment by PeterEbraert** [comment]
Hi guys,

Thanks for the feedback. 

I had already read that PDF. Not really what I was looking for. I would prefer to get a clean template for each kind of attestation.
Also, I would like to obtain legal directives. Are we allowed to print all the medical details (that are hidden behind the QR code) in plain text on the PDF or paper copy?

Any ideas?

**Comment by gabywh** [comment]
> Hi guys,
> 
> Thanks for the feedback.
> 
> I had already read that PDF. Not really what I was looking for. I would prefer to get a clean template for each kind of attestation.

UI Masks.
All I have seen so far are mock-ups as per in these docs.
Will ask around. 

> Also, I would like to obtain legal directives. Are we allowed to print all the medical details (that are hidden behind the QR code) in plain text on the PDF or paper copy?

The proposed EU regulation could be what you are looking for, also mentions "human readable" and paper versions, etc:
https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX:52021PC0130 (e.g. Point 14)

Other than that, best place to look for legal guidance your country's legal representative in the EU eHealthNetwork.

**Comment by rho-sk** [comment]
Also interested in this topic, we (SK) also plan to provide pdf (as print stable version) for non electronic use.

**Comment by PeterEbraert** [comment]
Find our working template for a vaccination attestation on http://peter.ebraert.be/Attest COVID-19 Vaccination EU 20210429A.pdf

-- In vitro data (QR code not in sync with data - and signed with a self generated key pair).

**Comment by rho-sk** [comment]
[01SKF2333D546F684533AF828A05CAC92D5C.pdf](https://github.com/ehn-digital-green-development/ehn-dgc-schema/files/6428243/01SKF2333D546F684533AF828A05CAC92D5C.pdf)
Here is our working template, multi lang version is implemented in way, that every language has separate page.

**Comment by gabywh** [comment]
UI layout has been quite an active topic of discussion these past couple of days within the eHealthNetwork, so I'll try to summarize here:
1. there are no EU-wide guidelines for UX/UI etc.
1. EU doesn't want to dictate to each Member State exactly how it must be, but to provide sample mock-ups etc as template / reference ideas for Member States
1. thus mock-ups are available in relevant docs (as above) to give some idea / reference
1. reference paper version of DGC has been worked on / discussed in EU eHealthNetwork, who will publish a version when formally approved 

Until then... here some general reference points we (NL) have been using, maybe of use to others to get started / refine their own ideas:
   * https://support.google.com/googleplay/android-developer/answer/9889712?hl=en
   * https://developer.apple.com/design/human-interface-guidelines/

**Comment by vitorpamplona** [comment]
> [01SKF2333D546F684533AF828A05CAC92D5C.pdf](https://github.com/ehn-digital-green-development/ehn-dgc-schema/files/6428243/01SKF2333D546F684533AF828A05CAC92D5C.pdf)
> Here is our working template, multi lang version is implemented in way, that every language has separate page.

Please keep in mind that the larger the QR, when printed on regular flimsy paper, the harder it is for the holder to hold it FLAT for a camera to have a successful read of the QR. DIVOC has a lot of issues on flatness of the paper in India. 

A smaller QR is easier for users to keep it flat. 

I would certainly recommend either printing in thicker paper or reducing the QR size.

**Comment by vitorpamplona** [comment]
To help with "template" ideas, we generally recommend a Business-Card-szed Tri-fold for multiple QRs: 

![Immunization Certificate - Business Card Size](https://user-images.githubusercontent.com/532031/117542674-91999600-afe7-11eb-98df-3c003e81d53d.jpg)
![Immunization Certificate - Business Card Size 2](https://user-images.githubusercontent.com/532031/117542676-92cac300-afe7-11eb-8100-c785f0174052.jpg)

**Comment by rho-sk** [comment]
> > [01SKF2333D546F684533AF828A05CAC92D5C.pdf](https://github.com/ehn-digital-green-development/ehn-dgc-schema/files/6428243/01SKF2333D546F684533AF828A05CAC92D5C.pdf)
> > Here is our working template, multi lang version is implemented in way, that every language has separate page.
> 
> @rho-sk Please keep in mind that the larger the QR, when printed on regular flimsy paper, the harder it is for the holder to hold it FLAT for a camera to have a successful read of the QR. DIVOC has a lot of issues on flatness of the paper in India.
> 
> A smaller QR is easier for users to keep it flat.
> 
> I would certainly recommend either printing in thicker paper or reducing the QR size.

When printing on A5 paper we achieve:
- easier printing at home
- at this size + the size of the QR code that we designed, the effective distance between the reader and the paper 1-1.5m which helps to comply with hygiene regulations
- the amount of data on the qr code makes them quite "dense" and at a small size they are easily damaged and on the contrary, they are worse legible or there must be at least a laser printer to maintain quality

and btw: creating a folding template on hard paper makes publishing more expensive

**Comment by vitorpamplona** [comment]
Yep, A5 was our first option as well. But we changed since no one has A5 at home. They ended up printing in A4 anyway. 

I did a quick test and My Pixel phone cannot read this QR at 1m away. It's too small. Much less at 1.5m. my phone needs to be at 40cm from the paper to read it. 

And agree, the compression mechanism of HCerts is not great.

**Comment by vitorpamplona** [comment]
Glad to see the suggested template is a foldable design on A4 paper: https://ec.europa.eu/health/sites/default/files/ehealth/docs/covid-certificate_paper_guidelines_en.pdf
